pkg/util/ipset/ipset.go

```diff
@@ -341,4 +484,49 @@ func IsNotFoundError(err error) bool {
 	return false
 }
 
+// checks if given protocol is supported in entry
+func validateProtocol(protocol string) bool {
+	if protocol == ProtocolTCP || protocol == ProtocolUDP {
+		return true
+	}
+	glog.Errorf("Invalid entry's protocol: %s, supported protocols are [%s, %s]", protocol, ProtocolTCP, ProtocolUDP)
+	return false
+}
+
+// parsePortRange parse the begin and end port from a raw string(format: a-b).  beginPort <= endPort
+// in the return value.
+func parsePortRange(portRange string) (beginPort int, endPort int, err error) {
```

Comment:
This code looks semi-fiddly, is there a unit test somewhere? I don't see it in this PR.

Reply:
Yes, I already added a unit test for this function. Please check: https://github.com/kubernetes/kubernetes/pull/57268/files#diff-1e6b0bd1e6429733c0d1633fe4741a98R799